examples/rsa/program/src/main.rs

```diff
@@ -3,47 +3,40 @@ sp1_zkvm::entrypoint!(main);
 
 use rsa::PaddingScheme;
 use rsa::PublicKey;
-use rsa::{
-    pkcs8::{DecodePrivateKey, DecodePublicKey},
-    RsaPrivateKey, RsaPublicKey,
-};
+use rsa::{pkcs8::DecodePublicKey, RsaPublicKey};
 use sha2::Digest;
 use sha2::Sha256;
-use std::vec;
-
-const RSA_2048_PRIV_DER: &[u8] = include_bytes!("rsa2048-priv.der");
-const RSA_2048_PUB_DER: &[u8] = include_bytes!("rsa2048-pub.der");
 
 pub fn main() {
-    let private_key = RsaPrivateKey::from_pkcs8_der(RSA_2048_PRIV_DER).unwrap();
-    let public_key = RsaPublicKey::from_public_key_der(RSA_2048_PUB_DER).unwrap();
-    println!("{:?}{:?}", private_key, public_key);
+    // Read an input to the program.
+    //
+    // Behind the scenes, this compiles down to a custom system call which handles reading inputs
+    let pk_der = sp1_zkvm::io::read::<Vec<u8>>();
```

Comment:
Does this program actually work? I don't think this should work because the io reader doesn't know how many bytes to actually read, if I'm not mistaken. You can either use `read_slice` and pass in a mutable slice with a particular length or you can just read a `RsaPublicKey` if that implements `Serde, Deserde`

Reply:
Well, it does when running in dev mode with `SP1Prover::execute(REGEX_IO_ELF, stdin)` but when trying with `SP1Prover::prove(REGEX_IO_ELF, stdin)` it prints the message from the program and gets killed:
![image](https://github.com/succinctlabs/sp1/assets/9572072/28796e42-8821-4ac8-9963-a07f13b776f7)

I thought it was something related to my machine (Ubuntu 22) because it gets killed too when running the initial RSA example.

> because the io reader doesn't know how many bytes to actually read

This was my understanding as well but then it works when `String` is passed along and `String` boils down to `Vec<u8>` :thinking: